webapp/src/actions/index.js

```diff
@@ -80,4 +80,34 @@ export function performSearch(terms) {
 
         return dispatch(searchPostsWithParams(teamId, {terms, is_or_search: false, include_deleted_channels: viewArchivedChannels, page: 0, per_page: 20}, true));
     };
-}
\ No newline at end of file
+}
+
+export function requeueItem(itemId) {
+    return async (dispatch, getState) => {
+        const command = `/agenda requeue ${itemId}`;
+        await clientExecuteCommand(dispatch, getState, command);
+        return {data: true};
+    };
+}
+
+export async function clientExecuteCommand(dispatch, getState, command) {
+    let currentChannel = getCurrentChannel(getState());
+    const currentTeamId = getCurrentTeamId(getState());
+
+    // Default to town square if there is no current channel (i.e., if Mattermost has not yet loaded)
+    if (!currentChannel) {
+        currentChannel = await Client4.getChannelByName(currentTeamId, 'town-square');
```

Comment:
Might be best to move the `town-square` into a variable. I remember this pattern being followed in some of the other plugins. @mickmister  what do you suggest ?

Reply:
If we are using a post dropdown item to requeue this agenda item, Mattermost has indeed loaded and we can assume `currentChannel` is already a valid channel. We don't need this if block

Reply:
All right.